code/game/objects/items/defib.dm

```diff
@@ -604,7 +604,10 @@
 						var/overall_damage = total_brute + total_burn + H.getToxLoss() + H.getOxyLoss()
 						var/mobhealth = H.health
 						H.adjustOxyLoss((mobhealth - HALFWAYCRITDEATH) * (H.getOxyLoss() / overall_damage), 0)
-						H.adjustToxLoss((mobhealth - HALFWAYCRITDEATH) * (H.getToxLoss() / overall_damage), 0)
+						if(HAS_TRAIT(H, TRAIT_TOXINLOVER))
+							H.adjustToxLoss((mobhealth - HALFWAYCRITDEATH) * (H.getToxLoss() / overall_damage), 0, TRUE) // force tox heal for toxin lovers
+						else
+							H.adjustToxLoss((mobhealth - HALFWAYCRITDEATH) * (H.getToxLoss() / overall_damage), 0)
```

Comment:
if you're using `force = TRUE` you can do away with the check for toxinlover above

`force` var on adjusttoxloss is specifically designed to deal with toxin lovers, it checks for it in the adjusttoxloss proc itself, it won't change the outcome if called on a non-toxinlover

Reply:
damn it, someone beat me